gettext-tools/man/msgcat.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.24.
.TH MSGCAT "1" "December 2015" "GNU gettext-tools 0.19.7" GNU
.SH NAME
msgcat \- combines several message catalogs
.SH SYNOPSIS
.B msgcat
[\fIOPTION\fR] [\fIINPUTFILE\fR]...
.SH DESCRIPTION
.\" Add any additional description here
.PP
Concatenates and merges the specified PO files.
Find messages which are common to two or more of the specified PO files.
By using the \fB\-\-more\-than\fR option, greater commonality may be requested
before messages are printed.  Conversely, the \fB\-\-less\-than\fR option may be
used to specify less commonality before messages are printed (i.e.
\fB\-\-less\-than\fR=\fI2\fR will only print the unique messages).  Translations,
comments, extracted comments, and file positions will be cumulated, except
that if \fB\-\-use\-first\fR is specified, they will be taken from the first PO file
to define them.
.PP
Mandatory arguments to long options are mandatory for short options too.
.SS "Input file location:"
.TP
INPUTFILE ...
input files
.TP
\fB\-f\fR, \fB\-\-files\-from\fR=\fIFILE\fR
get list of input files from FILE
.TP
\fB\-D\fR, \fB\-\-directory\fR=\fIDIRECTORY\fR
add DIRECTORY to list for input files search
.PP
If input file is -, standard input is read.
.SS "Output file location:"
.TP
\fB\-o\fR, \fB\-\-output\-file\fR=\fIFILE\fR
write output to specified file
.PP
The results are written to standard output if no output file is specified
or if it is -.
.SS "Message selection:"
.TP
-<, \fB\-\-less\-than\fR=\fINUMBER\fR
print messages with less than this many
definitions, defaults to infinite if not set
.TP
->, \fB\-\-more\-than\fR=\fINUMBER\fR
print messages with more than this many
definitions, defaults to 0 if not set
.TP
\fB\-u\fR, \fB\-\-unique\fR
shorthand for \fB\-\-less\-than\fR=\fI2\fR, requests
that only unique messages be printed
.SS "Input file syntax:"
.TP
\fB\-P\fR, \fB\-\-properties\-input\fR
input files are in Java .properties syntax
.TP
\fB\-\-stringtable\-input\fR
input files are in NeXTstep/GNUstep .strings
syntax
.SS "Output details:"
.TP
\fB\-t\fR, \fB\-\-to\-code\fR=\fINAME\fR
encoding for output
.TP
\fB\-\-use\-first\fR
use first available translation for each
message, don't merge several translations
.TP
\fB\-\-lang\fR=\fICATALOGNAME\fR
set 'Language' field in the header entry
.TP
\fB\-\-color\fR
use colors and other text attributes always
.TP
\fB\-\-color\fR=\fIWHEN\fR
use colors and other text attributes if WHEN.
WHEN may be 'always', 'never', 'auto', or 'html'.
.TP
\fB\-\-style\fR=\fISTYLEFILE\fR
specify CSS style rule file for \fB\-\-color\fR
.TP
\fB\-e\fR, \fB\-\-no\-escape\fR
do not use C escapes in output (default)
.TP
\fB\-E\fR, \fB\-\-escape\fR
use C escapes in output, no extended chars
.TP
\fB\-\-force\-po\fR
write PO file even if empty
.TP
\fB\-i\fR, \fB\-\-indent\fR
write the .po file using indented style
.TP
\fB\-\-no\-location\fR
do not write '#: filename:line' lines
.TP
\fB\-n\fR, \fB\-\-add\-location\fR
generate '#: filename:line' lines (default)
.TP
\fB\-\-strict\fR
write out strict Uniforum conforming .po file
.TP
\fB\-p\fR, \fB\-\-properties\-output\fR
write out a Java .properties file
.TP
\fB\-\-stringtable\-output\fR
write out a NeXTstep/GNUstep .strings file
.TP
\fB\-w\fR, \fB\-\-width\fR=\fINUMBER\fR
set output page width
.TP
\fB\-\-no\-wrap\fR
do not break long message lines, longer than
the output page width, into several lines
.TP
\fB\-s\fR, \fB\-\-sort\-output\fR
generate sorted output
.TP
\fB\-F\fR, \fB\-\-sort\-by\-file\fR
sort output by file location
.SS "Informative output:"
.TP
\fB\-h\fR, \fB\-\-help\fR
display this help and exit
.TP
\fB\-V\fR, \fB\-\-version\fR
output version information and exit
.SH AUTHOR
Written by Bruno Haible.
.SH "REPORTING BUGS"
Report bugs to <bug-gnu-gettext@gnu.org>.
.SH COPYRIGHT
Copyright \(co 2001-2010 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
.br
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
.SH "SEE ALSO"
The full documentation for
.B msgcat
is maintained as a Texinfo manual.  If the
.B info
and
.B msgcat
programs are properly installed at your site, the command
.IP
.B info msgcat
.PP
should give you access to the complete manual.


NEWS
Version 0.19.7 - December 2015

* Programming languages support:
  - XML:
    xgettext can now load custom string extraction rules supplied by
    consumer projects.  The rules are written in XML, utilizing the
    Internationalization Tag Set (ITS) standard.  All the existing
    XML-based language scanners (Glade, GSettings, and AppData) are
    rewritten using ITS.  In addition, msgfmt now has --xml option to
    merge translations back to the original XML document.

* Portability:
  - Improve OS/2 kLIBC support (still not complete)
  - Remove dependency on expat

Version 0.19.6 - September 2015

* Programming languages support:
  - AppData:
    xgettext now supports AppData file format, used by software center
    applications (e.g., GNOME Software) to describe installable
    applications.

* A new macro AM_GNU_GETTEXT_REQUIRE_VERSION can be used to indicate
  autopoint to pull the latest available infrastructure, instead of
  the exact version specified with AM_GNU_GETTEXT_VERSION.  When
  AM_GNU_GETTEXT_REQUIRE_VERSION is used, AM_GNU_GETTEXT_VERSION is
  ignored.

* po/Makefile.in.in can now insert the file $(DOMAIN).pot-header to
  $(DOMAIN).pot, instead of the standard header comments.

* Bug fixes:
  - Fix mishandling of gettext version numbers for minor releases, in
    po-mode.el and gettextize.
  - Fix build with --enable-relocatable.

Version 0.19.5 - July 2015

* xgettext now has a feature to perform syntax checks on msgid, which
  could enforce common styles of translatable strings, such as to
  prefer Unicode characters to the corresponding ASCII characters.
  They can be enabled with --check option or special "xgettext: "
  comment in the source code.  By default, no syntax checks are
  enabled.

* msgfilter and msgexec now have an option --newline, which appends a
  newline character to filter input and trims it from the filter
  output.  This would allow filter programs to be more POSIX friendly.

* The base Unicode standard is now updated to 8.0.0.  This
  particularly improves "\N{...}" notation handling of xgettext for
  Perl and Python.

* msginit is now capable of generating "Plural-Forms:" from Unicode
  CLDR.  This feature is still experimental, but you can try it by
  setting the GETTEXTCLDRDIR environment variable pointing to the
  directory where the CLDR archive is extracted.  The actual
  conversion is done by a helper program 'cldr-plural', which can be
  used as a generic converter and evaluator of CLDR plural forms.

* Programming languages support:
  - C++ with KDE: xgettext and msgfmt can now recognize KUIT (KDE User
    Interface Text) markup.  See the documentation section "KUIT
    Format Strings" for more info.
  - C++ with KDE: xgettext now recognizes all default KDE keywords.
    This removes the need for a long list of --keyword and --flag
    options to perform a reasonable extraction.

* Bug fixes:
  - xgettext C++11 raw string recognition is now stricter and don't
    accept unbalanced delimiters.
  - Suppress baseless warnings which msgfmt emits when processing a
    .desktop file.
  - xgettext line wrapping behaviour is now consistent between comment
    lines and non-comment lines.
  - Fix msgfilter-7 test failure on some platforms.
  - Fix VPATH build.

Version 0.19.4 - December 2014

* The --keyword option of xgettext now accepts same argument number
  for both singular and plural forms.

* Programming languages support:
  - C#: xgettext now properly handles Unicode characters encoded with
    surrogate pairs.
  - C/C++: xgettext now recognizes ISO/IEC 9899:2011 string literals
    prefixed by R, u8, u8R, u, uR, U, UR, L, or LR.
  - Shell: xgettext now properly recognizes Bash ANSI-C quoting ($'...').

* Bug fixes:
  - Fix integer overflow when reading certain MO files with msgunfmt.
  - Avoid invalid memory access in various cases.  In particular, when
    the same argument number is specified for singular/plural
    arguments, and when checking Lisp and Scheme format strings.

* Portability:
  - Building on Mac OS X 10.10 and AIX 7.1 is now supported.

Version 0.19.3 - October 2014

* Bug fixes:
  - Fix xgettext mishandling of octal character escapes in C.
  - Fix autopoint infinite recursion with certain configure.ac.

* The po/Makevars file has a new field MSGINIT_OPTIONS, that can be
  used to adjust msginit's operation.  This is particularly useful for
  controlling line wrapping behavior together with MSGMERGE_OPTIONS
  and XGETTEXT_OPTIONS.

* Portability:
  - Building on Solaris 10 and 11 with Solaris Studio compiler is now
    fixed.

Version 0.19.2 - July 2014

* Bug fixes:
  - Fix xgettext crash in parsing empty string literals in C and Vala.
  - ChangeLog file is added back to the gettext infrastructure.  It was
    mistakenly removed in 0.19.
  - Autoconf macro trace in autopoint now works again with Autoconf 2.68
    or earlier.  It was a regression in 0.19.

Version 0.19.1 - June 2014

* Programming languages support:

  - Desktop Entry:
    msgfmt now always reads the po/LINGUAS file, regardless of whether
    the LINGUAS environment variable is set.  The variable can now be
    used to restrict the languages list read from the po/LINGUAS file.

  - Vala:
    Bug fix in xgettext handling of "//" in string literals.  This was
    a regression after the C-99 Unicode escape support.

* The po/Makevars.template file now contains the newly added variables.

* msgfmt now treats errors in the PO file header as non-fatal.  Since
  0.19 msgfmt started to abort on the fatal errors, but some
  translation systems are still not ready to supply valid headers.

* Future backward-incompatibilities:
  - In future Gettext versions, msgfmt will treat header errors as
    fatal and terminate the command execution.

Version 0.19 - June 2014

* Programming languages support:

  - Desktop Entry:
    xgettext and msgfmt now support .desktop files, used by desktop
    applications, as input and output.

  - GSettings:

gettext-tools/src/msgcat.c
/* Concatenates several translation catalogs.
   Copyright (C) 2001-2007, 2009-2012 Free Software Foundation, Inc.
   Written by Bruno Haible <haible@clisp.cons.org>, 2001.

   This program is free software: you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 3 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program.  If not, see <http://www.gnu.org/licenses/>.  */


#ifdef HAVE_CONFIG_H
# include "config.h"
#endif

#include <getopt.h>
#include <limits.h>
#include <stdio.h>
#include <stdlib.h>
#include <locale.h>

#include "closeout.h"
#include "dir-list.h"
#include "str-list.h"
#include "file-list.h"
#include "error.h"
#include "error-progname.h"
#include "progname.h"
#include "relocatable.h"
#include "basename.h"
#include "message.h"
#include "read-catalog.h"
#include "read-po.h"
#include "read-properties.h"
#include "read-stringtable.h"
#include "write-catalog.h"
#include "write-po.h"
#include "write-properties.h"
#include "write-stringtable.h"
#include "color.h"
#include "msgl-cat.h"
#include "msgl-header.h"
#include "propername.h"
#include "gettext.h"

#define _(str) gettext (str)


/* Force output of PO file even if empty.  */
static int force_po;

/* Target encoding.  */
static const char *to_code;

/* Long options.  */
static const struct option long_options[] =
{
  { "add-location", no_argument, &line_comment, 1 },
  { "color", optional_argument, NULL, CHAR_MAX + 5 },
  { "directory", required_argument, NULL, 'D' },
  { "escape", no_argument, NULL, 'E' },
  { "files-from", required_argument, NULL, 'f' },
  { "force-po", no_argument, &force_po, 1 },
  { "help", no_argument, NULL, 'h' },
  { "indent", no_argument, NULL, 'i' },
  { "lang", required_argument, NULL, CHAR_MAX + 7 },
  { "no-escape", no_argument, NULL, 'e' },
  { "no-location", no_argument, &line_comment, 0 },
  { "no-wrap", no_argument, NULL, CHAR_MAX + 2 },
  { "output-file", required_argument, NULL, 'o' },
  { "properties-input", no_argument, NULL, 'P' },
  { "properties-output", no_argument, NULL, 'p' },
  { "sort-by-file", no_argument, NULL, 'F' },
  { "sort-output", no_argument, NULL, 's' },
  { "strict", no_argument, NULL, 'S' },
  { "stringtable-input", no_argument, NULL, CHAR_MAX + 3 },
  { "stringtable-output", no_argument, NULL, CHAR_MAX + 4 },
  { "style", required_argument, NULL, CHAR_MAX + 6 },
  { "to-code", required_argument, NULL, 't' },
  { "unique", no_argument, NULL, 'u' },
  { "use-first", no_argument, NULL, CHAR_MAX + 1 },
  { "version", no_argument, NULL, 'V' },
  { "width", required_argument, NULL, 'w', },
  { "more-than", required_argument, NULL, '>', },
  { "less-than", required_argument, NULL, '<', },
  { NULL, 0, NULL, 0 }
};


/* Forward declaration of local functions.  */
static void usage (int status)
#if defined __GNUC__ && ((__GNUC__ == 2 && __GNUC_MINOR__ >= 5) || __GNUC__ > 2)
        __attribute__ ((noreturn))
#endif
;


int
main (int argc, char **argv)
{
  int cnt;
  int optchar;
  bool do_help;
  bool do_version;
  char *output_file;
  const char *files_from;
  string_list_ty *file_list;
  msgdomain_list_ty *result;
  catalog_input_format_ty input_syntax = &input_format_po;
  catalog_output_format_ty output_syntax = &output_format_po;
  bool sort_by_msgid = false;
  bool sort_by_filepos = false;
  /* Language (ISO-639 code) and optional territory (ISO-3166 code).  */
  const char *catalogname = NULL;

  /* Set program name for messages.  */
  set_program_name (argv[0]);
  error_print_progname = maybe_print_progname;

#ifdef HAVE_SETLOCALE
  /* Set locale via LC_ALL.  */
  setlocale (LC_ALL, "");
#endif

  /* Set the text message domain.  */
  bindtextdomain (PACKAGE, relocate (LOCALEDIR));
  bindtextdomain ("bison-runtime", relocate (BISON_LOCALEDIR));
  textdomain (PACKAGE);

  /* Ensure that write errors on stdout are detected.  */
  atexit (close_stdout);

  /* Set default values for variables.  */
  do_help = false;
  do_version = false;
  output_file = NULL;
  files_from = NULL;
  more_than = 0;
  less_than = INT_MAX;
  use_first = false;

  while ((optchar = getopt_long (argc, argv, "<:>:D:eEf:Fhino:pPst:uVw:",
                                 long_options, NULL)) != EOF)
    switch (optchar)
      {
      case '\0':                /* Long option.  */
        break;

      case '>':
        {
          int value;
          char *endp;
          value = strtol (optarg, &endp, 10);
          if (endp != optarg)
            more_than = value;
        }
        break;

      case '<':
        {
          int value;
          char *endp;
          value = strtol (optarg, &endp, 10);
          if (endp != optarg)
            less_than = value;
        }
        break;

      case 'D':
        dir_list_append (optarg);
        break;

      case 'e':
        message_print_style_escape (false);
        break;

      case 'E':
        message_print_style_escape (true);
        break;

      case 'f':
        files_from = optarg;
        break;

      case 'F':
        sort_by_filepos = true;
        break;

      case 'h':
        do_help = true;
        break;

      case 'i':
        message_print_style_indent ();
        break;

      case 'n':
        line_comment = 1;
        break;

      case 'o':
        output_file = optarg;
        break;

      case 'p':
        output_syntax = &output_format_properties;
        break;

      case 'P':
        input_syntax = &input_format_properties;
        break;

      case 's':
        sort_by_msgid = true;
        break;

      case 'S':
        message_print_style_uniforum ();
        break;

      case 't':
        to_code = optarg;
        break;

      case 'u':
        less_than = 2;
        break;

      case 'V':
        do_version = true;
        break;

      case 'w':
        {
          int value;
          char *endp;
          value = strtol (optarg, &endp, 10);
          if (endp != optarg)
            message_page_width_set (value);
        }
        break;

      case CHAR_MAX + 1:
        use_first = true;
        break;

      case CHAR_MAX + 2: /* --no-wrap */
        message_page_width_ignore ();
        break;

      case CHAR_MAX + 3: /* --stringtable-input */
        input_syntax = &input_format_stringtable;
        break;

      case CHAR_MAX + 4: /* --stringtable-output */
        output_syntax = &output_format_stringtable;
        break;

      case CHAR_MAX + 5: /* --color */
        if (handle_color_option (optarg))
          usage (EXIT_FAILURE);
        break;

      case CHAR_MAX + 6: /* --style */
        handle_style_option (optarg);
        break;

      case CHAR_MAX + 7: /* --lang */
        catalogname = optarg;
        break;

      default:
        usage (EXIT_FAILURE);
        /* NOTREACHED */
      }

  /* Version information requested.  */
  if (do_version)
    {
      printf ("%s (GNU %s) %s\n", basename (program_name), PACKAGE, VERSION);
      /* xgettext: no-wrap */
      printf (_("Copyright (C) %s Free Software Foundation, Inc.\n\
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>\n\
This is free software: you are free to change and redistribute it.\n\
There is NO WARRANTY, to the extent permitted by law.\n\
"),
              "2001-2010");
      printf (_("Written by %s.\n"), proper_name ("Bruno Haible"));
      exit (EXIT_SUCCESS);
    }

  /* Help is requested.  */
  if (do_help)
    usage (EXIT_SUCCESS);

  if (color_test_mode)
    {
      print_color_test ();
      exit (EXIT_SUCCESS);
    }

  /* Verify selected options.  */
  if (!line_comment && sort_by_filepos)
    error (EXIT_FAILURE, 0, _("%s and %s are mutually exclusive"),
           "--no-location", "--sort-by-file");

  if (sort_by_msgid && sort_by_filepos)
    error (EXIT_FAILURE, 0, _("%s and %s are mutually exclusive"),
           "--sort-output", "--sort-by-file");

  /* Check the message selection criteria for sanity.  */
  if (more_than >= less_than || less_than < 2)
    error (EXIT_FAILURE, 0,
           _("impossible selection criteria specified (%d < n < %d)"),
           more_than, less_than);

  /* Determine list of files we have to process.  */
  if (files_from != NULL)
    file_list = read_names_from_file (files_from);
  else
    file_list = string_list_alloc ();
  /* Append names from command line.  */
  for (cnt = optind; cnt < argc; ++cnt)
    string_list_append_unique (file_list, argv[cnt]);

  /* Read input files, then filter, convert and merge messages.  */
  result =
    catenate_msgdomain_list (file_list, input_syntax,
                             output_syntax->requires_utf8 ? "UTF-8" : to_code);

  string_list_free (file_list);

  /* Sorting the list of messages.  */
  if (sort_by_filepos)
    msgdomain_list_sort_by_filepos (result);
  else if (sort_by_msgid)
    msgdomain_list_sort_by_msgid (result);

  /* Set the Language field in the header.  */
  if (catalogname != NULL)
    msgdomain_list_set_header_field (result, "Language:", catalogname);

  /* Write the PO file.  */
  msgdomain_list_print (result, output_file, output_syntax, force_po, false);

  exit (EXIT_SUCCESS);
}


/* Display usage information and exit.  */
static void
usage (int status)
{
  if (status != EXIT_SUCCESS)
    fprintf (stderr, _("Try '%s --help' for more information.\n"),
             program_name);
  else
    {
      printf (_("\
Usage: %s [OPTION] [INPUTFILE]...\n\
"), program_name);
      printf ("\n");
      /* xgettext: no-wrap */
      printf (_("\
Concatenates and merges the specified PO files.\n\
Find messages which are common to two or more of the specified PO files.\n\
By using the --more-than option, greater commonality may be requested\n\
before messages are printed.  Conversely, the --less-than option may be\n\
used to specify less commonality before messages are printed (i.e.\n\
--less-than=2 will only print the unique messages).  Translations,\n\
comments, extracted comments, and file positions will be cumulated, except\n\
that if --use-first is specified, they will be taken from the first PO file\n\
to define them.\n\
"));
      printf ("\n");
      printf (_("\
Mandatory arguments to long options are mandatory for short options too.\n"));
      printf ("\n");
      printf (_("\
Input file location:\n"));
      printf (_("\
  INPUTFILE ...               input files\n"));
      printf (_("\
  -f, --files-from=FILE       get list of input files from FILE\n"));
      printf (_("\
  -D, --directory=DIRECTORY   add DIRECTORY to list for input files search\n"));
      printf (_("\
If input file is -, standard input is read.\n"));
      printf ("\n");
      printf (_("\
Output file location:\n"));
      printf (_("\
  -o, --output-file=FILE      write output to specified file\n"));
      printf (_("\
The results are written to standard output if no output file is specified\n\
or if it is -.\n"));
      printf ("\n");
      printf (_("\
Message selection:\n"));
      printf (_("\
  -<, --less-than=NUMBER      print messages with less than this many\n\
                              definitions, defaults to infinite if not set\n"));
      printf (_("\
  ->, --more-than=NUMBER      print messages with more than this many\n\
                              definitions, defaults to 0 if not set\n"));
      printf (_("\
  -u, --unique                shorthand for --less-than=2, requests\n\
                              that only unique messages be printed\n"));
      printf ("\n");
      printf (_("\
Input file syntax:\n"));
      printf (_("\
  -P, --properties-input      input files are in Java .properties syntax\n"));
      printf (_("\
      --stringtable-input     input files are in NeXTstep/GNUstep .strings\n\
                              syntax\n"));
      printf ("\n");
      printf (_("\
Output details:\n"));
      printf (_("\
  -t, --to-code=NAME          encoding for output\n"));
      printf (_("\
      --use-first             use first available translation for each\n\
                              message, don't merge several translations\n"));
      printf (_("\
      --lang=CATALOGNAME      set 'Language' field in the header entry\n"));
      printf (_("\
      --color                 use colors and other text attributes always\n\
      --color=WHEN            use colors and other text attributes if WHEN.\n\
                              WHEN may be 'always', 'never', 'auto', or 'html'.\n"));
      printf (_("\
      --style=STYLEFILE       specify CSS style rule file for --color\n"));
      printf (_("\
  -e, --no-escape             do not use C escapes in output (default)\n"));
      printf (_("\
  -E, --escape                use C escapes in output, no extended chars\n"));
      printf (_("\
      --force-po              write PO file even if empty\n"));
      printf (_("\
  -i, --indent                write the .po file using indented style\n"));
      printf (_("\
      --no-location           do not write '#: filename:line' lines\n"));
      printf (_("\
  -n, --add-location          generate '#: filename:line' lines (default)\n"));
      printf (_("\
      --strict                write out strict Uniforum conforming .po file\n"));
      printf (_("\
  -p, --properties-output     write out a Java .properties file\n"));
      printf (_("\
      --stringtable-output    write out a NeXTstep/GNUstep .strings file\n"));
      printf (_("\
  -w, --width=NUMBER          set output page width\n"));
      printf (_("\
      --no-wrap               do not break long message lines, longer than\n\
                              the output page width, into several lines\n"));
      printf (_("\
  -s, --sort-output           generate sorted output\n"));
      printf (_("\
  -F, --sort-by-file          sort output by file location\n"));
      printf ("\n");
      printf (_("\
Informative output:\n"));
      printf (_("\
  -h, --help                  display this help and exit\n"));
      printf (_("\
  -V, --version               output version information and exit\n"));
      printf ("\n");
      /* TRANSLATORS: The placeholder indicates the bug-reporting address
         for this package.  Please add _another line_ saying
         "Report translation bugs to <...>\n" with the address for translation
         bugs (typically your translation team's web or email address).  */
      fputs (_("Report bugs to <bug-gnu-gettext@gnu.org>.\n"),
             stdout);
    }

  exit (status);
}
